problem.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"math/rand"
+	"fmt"
+)
+
+type Problem struct {
+	problem string
+	solution string
+}
+
+func (p *Problem) checkSolution(solution string) bool {
+	return solution == p.solution
+}
+
+func makeProblem_EqAdd() *Problem {
```

Comment:
```suggestion
func additionProblem() *Problem {
```

Go typically uses camelCase for function names, and tries to use nouns for functions that return something. In this case, it could even be a kind of constructor.